### PR TITLE
add monitoring of tracks stopping source

### PR DESCRIPTION
**Pull request by mtosi**
as PR https://github.com/cms-sw/cmssw/pull/13628

this PR is meant to add monitoring plots for the stopping source

3 plots (1 1D and 2 2D) are added in the general properties directories
3 plots (1 1D and 2 2D) x each step of the iterative tracking are added in the TrackBuilding directory

the 1D plot has 10bins (one per source type)
the 2D plots give this info as function of the track eta and phi, respectively (using the usual eta and phi binning)

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1

File: DQM/TrackingMonitor/src/TrackAnalyzer.cc [hunk 1]
```diff
@@ -517,6 +522,42 @@ void TrackAnalyzer::bookHistosForHitProperties(DQMStore::IBooker & ibooker) {
 	algorithm->setBinLabel(ibin+1,reco::TrackBase::algoNames[ibin]);
         oriAlgo->setBinLabel(ibin+1,reco::TrackBase::algoNames[ibin]);
       }
+
+      // DataFormats/TrackReco/interface/TrajectoryStopReasons.h
+      std::vector<std::string> StopReasonName;
```
Comment (davidlange6) [inline DQM/TrackingMonitor/src/TrackAnalyzer.cc]: @mtosi - pls change to

std::vector<std::string> stopReasonName={"UNINITIALIZED","MAX_HITS"....};

thx.

File: DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc [hunk 2]
```diff
@@ -200,6 +204,45 @@ void TrackBuildingAnalyzer::initHisto(DQMStore::IBooker & ibooker)
     NumberOfRecHitsPerSeedVsEtaProfile->setAxisTitle("Seed #eta",1);
     NumberOfRecHitsPerSeedVsEtaProfile->setAxisTitle("Number of RecHits of each Seed",2);
   }
+
+  if (doAllTCPlots || doStopSource) {
+    // DataFormats/TrackReco/interface/TrajectoryStopReasons.h
+    std::vector<std::string> StopReasonName;
+    StopReasonName.push_back("UNINITIALIZED");
```
Comment (davidlange6) [inline DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc]: same comment as above.

**Comment by mtosi** [comment]
I've just committed the fix to the vector initialization, hope it is fine, now
I'm going to apply the same to PR https://github.com/cms-sw/cmssw/pull/13659

**Comment by davidlange6** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**